File: GARI_Umbria_2026-08-03/DotNet/AgronicaCore_2010/AgronicaCoreMetaSchemaDAL/Utenti_Impostazioni_FiltroMono.vb
Imports System.Data.OleDb
Imports AgronicaCoreDataProvider.UtilityProvider







'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§

Public Class Utenti_Impostazioni_FiltroMono_R
    Inherits AgronicaCoreDataProvider.DataProvider




    '##############################################################################################
    Public Function Leggi( _
                        ByVal Impostazione_Cod As Integer, _
                        ByVal ID_0 As Integer, _
                                ByVal xSelezioneVariabile As AgronicaCoreDataProvider.AgronicaCoreParametri.enumSelezioneVariabile, _
                                ByVal xFiltroAggiuntivo As String, _
                                ByVal xOrderBy As String, _
                                ByRef objParametri As AgronicaCoreDataProvider.AgronicaCoreParametri _
                                ) As DataTable

        Dim NomeRoutine As String = "AgronicaCoreMetaSchemaDAL.Utenti_Impostazioni_FiltroMono_R.Leggi()"

        '====================================================================================
        'Parametri opzionali :
        '   Veg_Cod = 0           =>  tutti i record
        '====================================================================================

        Dim MessaggioErrore As String = ""
        Dim StrSQL As New System.Text.StringBuilder
        Dim DT As DataTable

        Try

            Select Case xSelezioneVariabile

                Case AgronicaCoreDataProvider.AgronicaCoreParametri.enumSelezioneVariabile.Selezione_TabellaDatiMinimi
                    '//////////////////////////////////////////////////////////////////////
                    '//////////////////////////////////////////////////////////////////////


                Case AgronicaCoreDataProvider.AgronicaCoreParametri.enumSelezioneVariabile.Selezione_TabellaCompleta
                    '//////////////////////////////////////////////////////////////////////
                    '//////////////////////////////////////////////////////////////////////

                    '//////////////////////////////////////////////////////////////////////
                    '//////////////////////////////////////////////////////////////////////



                Case AgronicaCoreDataProvider.AgronicaCoreParametri.enumSelezioneVariabile.Selezione_JoinDescrizioni
                    '
                    StrSQL.Length = 0
                    StrSQL.Append(" SELECT      Utenti_Impostazioni_FiltroMono.Piva_SuperUser, Utenti_Impostazioni_FiltroMono.UserName, Utenti_Impostazioni_FiltroMono.Impostazione_Cod,   ")
                    StrSQL.Append("             Utenti_Impostazioni_FiltroMono.ID_0, Utenti_Impostazioni.Impostazione_Valore_1, Utenti_Impostazioni.Impostazione_Valore_2, ")
                    StrSQL.Append("             Utenti_Impostazioni.Impostazione_Valore_3, Utenti_Impostazioni.Impostazione_Valore_4 ")
                    StrSQL.Append(" FROM        Utenti_Impostazioni_FiltroMono ")
                    StrSQL.Append(" INNER JOIN  Utenti_Impostazioni ON ")
                    StrSQL.Append("             Utenti_Impostazioni.Piva_SuperUser = Utenti_Impostazioni_FiltroMono.Piva_SuperUser ")
                    StrSQL.Append("             AND  Utenti_Impostazioni.UserName = Utenti_Impostazioni_FiltroMono.UserName ")
                    StrSQL.Append("             AND Utenti_Impostazioni.Impostazione_Cod = Utenti_Impostazioni_FiltroMono.Impostazione_Cod ")
                    StrSQL.Append(" WHERE     1=1  ")
                    If objParametri.PivaSuperUser <> "" Then
                        StrSQL.Append(" AND     (Utenti_Impostazioni_FiltroMono.Piva_SuperUser = '" & Agro_SQL_SaveText(objParametri.PivaSuperUser) & "')   ")
                    End If

                    If objParametri.UsernameOperazione <> "" Then
                        StrSQL.Append(" AND     (Utenti_Impostazioni_FiltroMono.Username = '" & Agro_SQL_SaveText(objParametri.UtenteUsername) & "')   ")

                    End If

                    If Impostazione_Cod <> 0 Then
                        StrSQL.Append(" AND     (Utenti_Impostazioni_FiltroMono.Impostazione_Cod = " & Agro_SQL_SaveNum(Impostazione_Cod) & ")   ")
                    End If

                    If ID_0 <> 0 Then

                        StrSQL.Append(" AND     (Utenti_Impostazioni_FiltroMono.ID_0 = " & Agro_SQL_SaveNum(ID_0) & ")   ")

                    End If


                    '--------------------------------------------------------------------------
                    If xFiltroAggiuntivo <> "" Then
                        StrSQL.Append(" AND " & Agro_SQL_Save_xFiltroAggiuntivo(xFiltroAggiuntivo, , objParametri))
                    End If

                    '--------------------------------------------------------------------------
                    If xOrderBy <> "" Then
                        StrSQL.Append(" ORDER BY " & Agro_SQL_Save_xOrderBy(xOrderBy, objParametri))
                    Else
                        StrSQL.Append(" ORDER BY Utenti_Impostazioni_FiltroMono.Piva_SuperUser, Utenti_Impostazioni_FiltroMono.Username, Utenti_Impostazioni_FiltroMono.Impostazione_Cod ")

                    End If

                Case AgronicaCoreDataProvider.AgronicaCoreParametri.enumSelezioneVariabile.Selezione_JoinCompleta
                    '
                    '
                    '
                    '


            End Select



            '--------------------------------------------------------------------------
            DT = EseguiQuery_Lettura(objParametri, StrSQL.ToString, NomeRoutine)
            '--------------------------------------------------------------------------

        Catch ex As Exception
            MessaggioErrore = ex.Message
            Scrivi_LOG(objParametri, NomeRoutine, MessaggioErrore)
            DT = Nothing
            Throw New Exception("[" & NomeRoutine & "] : " & MessaggioErrore)
        End Try

        Return DT

    End Function





End Class


'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§
'§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§§

File: GARI_Umbria_2026-08-03/DotNet/AgronicaNetCore/AgronicaNetCore.Utenti.DAL/DataLayer/UtentiImpostazioniFiltroMono/UtentiImpostazioniFiltroMono.cs
using AgronicaNetCore.Base.Constants;
using AgronicaNetCore.Base.Models;
using AgronicaNetCore.Utenti.DAL.Resources;
using Microsoft.Extensions.Localization;
using System.Data;
using System.Text;

namespace AgronicaNetCore.Utenti.DAL.DataLayer.UtentiImpostazioniFiltroMono
{
    public class UtentiImpostazioniFiltroMono : BaseDALUtenti, IUtentiImpostazioniFiltroMono
    {
        public UtentiImpostazioniFiltroMono(IServiceProvider provider, IStringLocalizer<Messages> localizer) : base(provider, localizer) { }

        public async Task<DataTable> LeggiAsync(TipiEnumerativi.Enum_Impostazioni_Utenti Impostazione_Cod, int ID_0, AgronicaCoreParametriUtenti objParametriUtenti, AgronicaCoreParametriServer objParametriServer)
        {
            var stbQuery = new StringBuilder();
            var parametriSql = new Dictionary<string, object>();
            DataTable result;

            //corrisponde alla query effettuata dalla classe AgronicaCoreUtentiDAL.Utenti_Impostazioni_FiltroMono_R.Leggi con xSelezioneVariabile = AgronicaCoreParametri.enumSelezioneVariabile.Selezione_JoinDescrizioni
            stbQuery.AppendLine(" SELECT      Utenti_Impostazioni_FiltroMono.Piva_SuperUser, Utenti_Impostazioni_FiltroMono.UserName, Utenti_Impostazioni_FiltroMono.Impostazione_Cod,   ");
            stbQuery.AppendLine("             Utenti_Impostazioni_FiltroMono.ID_0, Utenti_Impostazioni.Impostazione_Valore_1, Utenti_Impostazioni.Impostazione_Valore_2, ");
            stbQuery.AppendLine("             Utenti_Impostazioni.Impostazione_Valore_3, Utenti_Impostazioni.Impostazione_Valore_4,Utenti_Impostazioni_FiltroMono.Str_0  ");
            stbQuery.AppendLine(" FROM        Utenti_Impostazioni_FiltroMono ");
            stbQuery.AppendLine(" INNER JOIN  Utenti_Impostazioni ON ");
            stbQuery.AppendLine("             Utenti_Impostazioni.Piva_SuperUser = Utenti_Impostazioni_FiltroMono.Piva_SuperUser ");
            stbQuery.AppendLine("             AND  Utenti_Impostazioni.UserName = Utenti_Impostazioni_FiltroMono.UserName ");
            stbQuery.AppendLine("             AND Utenti_Impostazioni.Impostazione_Cod = Utenti_Impostazioni_FiltroMono.Impostazione_Cod ");
            stbQuery.AppendLine(" WHERE       1=1  ");

            if (!string.IsNullOrEmpty(objParametriUtenti.PivaSuperUser))
            {
                stbQuery.AppendLine(" AND     Utenti_Impostazioni_FiltroMono.Piva_SuperUser = @pivaSuperUser");
                parametriSql.Add("@pivaSuperUser", objParametriUtenti.PivaSuperUser);
            }

            if (!string.IsNullOrEmpty(objParametriUtenti.UtenteUsername))
            {
                stbQuery.AppendLine(" AND     Utenti_Impostazioni_FiltroMono.Username = @username");
                parametriSql.Add("@username", objParametriUtenti.UtenteUsername);
            }

            if (Impostazione_Cod != 0)
            {
                stbQuery.AppendLine(" AND     Utenti_Impostazioni_FiltroMono.Impostazione_Cod = @impostazioneCod");
                parametriSql.Add("@impostazioneCod", Impostazione_Cod);
            }

            if (ID_0 != 0)
            {
                stbQuery.AppendLine(" AND     Utenti_Impostazioni_FiltroMono.ID_0 = @ID_0");
                parametriSql.Add("@ID_0", ID_0);
            }

            stbQuery.AppendLine(" ORDER BY    Utenti_Impostazioni_FiltroMono.Piva_SuperUser, Utenti_Impostazioni_FiltroMono.Username, Utenti_Impostazioni_FiltroMono.Impostazione_Cod ");

            try
            {
                result = await GetDataProvider(objParametriUtenti).ExecuteReadAsync(stbQuery.ToString(), parametriSql);
            }
            catch (Exception ex)
            {
                LogError(ex.Message, objParametriServer, ex);
                throw;
            }

            return result;
        }
    }
}
